### PR TITLE
[4.1.x][6921] SingleFileUpload component enhancements

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6921

### CODE REVIEW

File: ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx [hunk 1]
```diff
@@ -34,6 +34,11 @@ import Button from '@mui/material/Button';
 import useSiteUIConfig from '../../hooks/useSiteUIConfig';
 import { ensureSingleSlash } from '../../utils/string';
 import { toQueryString } from '../../utils/object';
+import { getResponseError } from '../UploadDialog/util';
+import Tooltip from '@mui/material/Tooltip';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
```
Comment (rart) [inline ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx]:
```suggestion
import CloseIcon from '@mui/icons-material/CloseRounded';
```

File: ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx [hunk 2]
```diff
@@ -342,14 +355,35 @@ export function SingleFileUpload(props: SingleFileUploadProps) {
...
+        {error ? (
+          <Alert
+            icon={false}
+            severity="error"
+            action={
+              <Tooltip title={<FormattedMessage defaultMessage="Dismiss" />}>
+                <IconButton onClick={() => resetState()} size="small">
+                  <CloseIcon />
+                </IconButton>
+              </Tooltip>
```
Comment (rart) [inline ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx]: Can we have some sort of retry logic? If users wish to chose a different file, they can use the regular button. Make sure that clears the error too.
```suggestion
              <Tooltip title={<FormattedMessage defaultMessage="Retry" />}>
                <IconButton onClick={() => resetState()} size="small">
                  <ReplayRoundedIcon />
                </IconButton>
              </Tooltip>
```

File: ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx [hunk 3]
```diff
@@ -342,14 +355,35 @@ export function SingleFileUpload(props: SingleFileUploadProps) {
       </form>
       <div className="uppy-progress-bar" />
       <div className="uploaded-files">
-        <Typography variant="subtitle1" component="h2" className={classes.description}>
```
Comment (rart) [inline ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx:345]: Looks like classes.description is no longer in use. Delete.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx (hunk 3 above)
Comment (jvega190) [reply ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx:345]: yeah, already removed https://github.com/craftercms/studio-ui/pull/3991/files#diff-97d6b37d1903dc757211f4bd460923f9a31f8bae376739fde8027950f764efa6L76

**Review by rart — APPROVED** [review]

**Pull request merged**